### PR TITLE
Add inspector analytics

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3960.

### CODE REVIEW

File: gui/src/app/components/inspector/expose/expose.js [hunk 1]
```diff
@@ -29,6 +38,7 @@ class InspectorExpose extends React.Component {
         }
       }
     });
+    this.analytics.addCategory('Unit').sendEvent(this.props.analytics.CLICK);
```
Comment (hatched) [inline gui/src/app/components/inspector/expose/expose.js]: maybe this should be 'Unit Details'?

File: gui/src/app/components/inspector/relations/relations.js [hunk 2]
```diff
@@ -29,6 +34,7 @@ class InspectorRelations extends React.Component {
           id: this.props.service.get('id'),
           activeComponent: 'relate-to'
         }}});
+    this.analytics.addCategory('Build A Relation').sendEvent(this.props.analytics.CLICK);
```
Comment (hatched) [inline gui/src/app/components/inspector/relations/relations.js]: Maybe just 'Build' or 'Create' ?

File: gui/src/app/init/relation-utils.js [hunk 3]
```diff
@@ -531,7 +531,7 @@ function getAvailableEndpoints(
     return relatableEndpoints;
   }
   // Filter out the existing relations
-  return relatableEndpoints.filter(relation =>
+  return relatableEndpoints && relatableEndpoints.filter(relation =>
```
Comment (hatched) [inline gui/src/app/init/relation-utils.js]: What's up with this? Should we add a test for this issue?

Comment (huwshimi) [reply gui/src/app/init/relation-utils.js]: This was a bug that I found while testing the events. I'd add a test, but the tests require YUI, so they're no able to be run.

Comment (hatched) [reply gui/src/app/init/relation-utils.js]: Ok that's fine you can leve it as is

**Pull request merged**